Crash because of FromJSON

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

We used the Fuzz engine to modify some Test(TestFromJSON) data, and then the following crash appeared. We debugged and eventually found the crash point is at line 60 of `theme/json.go`. We feel that this could be an out-of-boundary bug.

### How to reproduce

The simplified test code is in **Example code**

### Screenshots

Crash log
```
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1

goroutine 19 [running]:
testing.tRunner.func1.2({0x6fd6a0, 0xc0000a2810})
	D:/.../go1.19/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	D:/.../go1.19/src/testing/testing.go:1399 +0x39f
panic({0x6fd6a0, 0xc0000a2810})
	D:/.../go1.19/src/runtime/panic.go:884 +0x212
fyne.io/fyne/v2/theme.hexColor.color({0xc00009ff28?, 0xc0002a2270?})
	D:/.../fyne/theme/json.go:60 +0x477

fyne.io/fyne/v2/theme.(*jsonTheme).Color(0xc00009c450, {0x71c7d9, 0xa}, 0x1)
	D:/.../fyne/theme/json.go:133 +0x12f
fyne.io/fyne/v2/theme.TestFromJSON2(0x0?)
	D:/.../fyne/theme/FromJSON_fuzz_test.go  :34 +0x5f
testing.tRunner(0xc000085040, 0x750c40)
	D:/.../go1.19/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	D:/.../go1.19/src/testing/testing.go:1493 +0x35f
```

### Example code

```go
repository.Register("file", intRepo.NewFileRepository()) // file uri resolving (avoid test import loop)
th, _ := FromJSON("{\"Colors\":{\"foreground\":\"\xb1\"}}")
th.Color(ColorNameForeground, VariantLight)
```

### Fyne version

commit [cfb88d7](https://github.com/fyne-io/fyne/commit/cfb88d7c4038f737f94ca702f7cf4e5d6c86e01d)

### Go compiler version

1.19

### Operating system and version

Windows 10

### Additional Information

_No response_

--- a/theme/json.go
+++ b/theme/json.go
@@ -39,7 +39,7 @@ type hexColor string
 
 func (h hexColor) color() (color.Color, error) {
 	data := h
-	switch len(h) {
+	switch len([]rune(h)) {
 	case 8, 6:
 	case 9, 7: // remove # prefix
 		data = h[1:]

--- a/theme/json_test.go
+++ b/theme/json_test.go
@@ -28,6 +28,10 @@ func TestFromJSON(t *testing.T) {
 	assert.Equal(t, float32(5), th.Size(SizeNameInlineIcon))
 	assert.Equal(t, "NotoMono-Regular.ttf", th.Font(fyne.TextStyle{Monospace: true}).Name())
 	assert.Equal(t, "cancel_Paths.svg", th.Icon(IconNameCancel).Name())
+
+	th, _ = FromJSON("{\"Colors\":{\"foreground\":\"\xb1\"}}")
+	c := th.Color(ColorNameForeground, VariantLight)
+	assert.NotNil(t, c)
 }
 
 func TestFromTOML_Resource(t *testing.T) {

